docs(flutter): Update sentry dart plugin docs (#10354)
getsentry/sentry-docs: docs/platforms/flutter/upload-debug.mdx

To modify this configuration, add a `sentry:` configuration to the end of your `pubspec.yaml` file:

```yaml

>   upload_debug_symbols: true

Reviewer: don't we need the sentry field here?

`release`

> The release version for source maps, it should match the release set by the SDK. This is a `string` type with default value `name@version` from pubspec. If a build number is included in the version, it is utilized as dist. The alternative environmental variable is `SENTRY_RELEASE`.

Reviewer 2: ```suggestion
The release version for source maps, which should match the release set by the SDK. This is a `string` type with default value `name@version` from pubspec. If a build number is included in the version, it is utilized as dist. The alternative environmental variable is `SENTRY_RELEASE`.
```

Setting a custom release can be specified via an environment variable or plugin configuration. Once set, it is used as is without further modification.

`dist`

> Custom dist can also be set via environment variable `SENTRY_DIST` or plugin configuration. It replaces or adds to the build number in the default release.

Reviewer 2: ```suggestion
Custom dist can also be set via the environment variable `SENTRY_DIST` or via plugin configuration. It replaces or adds to the build number in the default release.
```

Ignore missing commits previously used in the release. This is a boolean type with default value `false`.

`bin_dir`

The folder where the plugin downloads the sentry-cli binary. This is a `string` type with default value `.dart_tool/pub/bin/sentry_dart_plugin`.

`bin_path`

> Path to the sentry-cli binary to use instead of downloading. Make sure to use the correct version. This is a `string` type and is empty per default.

Reviewer 2: ```suggestion
Path to the sentry-cli binary to use instead of downloading. Make sure to use the correct version. This is a `string` type and is empty by default.
```

`sentry_cli_cdn_url`

> Alternative place to download sentry-cli. This is a `string` type with default value `https://downloads.sentry-cdn.com/sentry-cli`. Alternativeley, this can also be set with `SENTRYCLI_CDNURL` environment variable.

Reviewer 2: ```suggestion
Alternative place to download sentry-cli. This is a `string` type with default value `https://downloads.sentry-cdn.com/sentry-cli`. Alternatively, this can also be set with the `SENTRYCLI_CDNURL` environment variable.
```
